### ISSUE TITLE
[Bug]: Unable to Use Subtitle doc block in MDX without a Story

### ISSUE THREAD

**Issue by peschee** [storybookjs/storybook#27139]
### Describe the bug

Since `8.1.0`, the `<Subtitle>` doc block seems to be broken in MDX.

Before, the following used to work:

```mdx
import { Meta, Subtitle } from '@storybook/blocks';

<Meta title="Some/Title" />

<Subtitle>Some subtitle</Subtitle>
```

now, the following error is shown:

`No CSF file attached to this docs file, did you forget to use <Meta of={} />?`

Removing the `<Subtitle>` doc block fixes things…

### To Reproduce

https://stackblitz.com/edit/github-kqs5ug?file=src%2Fstories%2FConfigure.mdx

### System

```bash
Storybook Environment Info:

  System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm <----- active
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    @storybook/addon-essentials: ^8.1.0-beta.1 => 8.1.0 
    @storybook/addon-links: ^8.1.0-beta.1 => 8.1.0 
    @storybook/blocks: ^8.1.0-beta.1 => 8.1.0 
    @storybook/test: ^8.1.0-beta.1 => 8.1.0 
    @storybook/web-components: ^8.1.0-beta.1 => 8.1.0 
    @storybook/web-components-vite: ^8.1.0-beta.1 => 8.1.0 
    storybook: ^8.1.0-beta.1 => 8.1.0
```

### Additional context

Also happens with `8.1.0`. The minimal repro was created from storybook.new, which still seems to use the beta.

**Comment by peschee** [comment]
I just noticed that after downgrading our SB instance back to `8.0.10` (after regenerating the lock file), it does not work any more even there. So it might be an issue with a transitive dependency?

**Comment by shilman** [comment]
Probably due to https://github.com/storybookjs/storybook/pull/22552

**Comment by peschee** [comment]
Cool @shilman! Will this be released in a next `8.1.x` version or also backported as a bugfix to previous versions?

**Comment by shilman** [comment]
This will be released in 8.1.1, probably within the next hour.
